libs/designsystem/src/lib/components/progress-circle/progress-circle.component.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  Component,
+  Input,
+  ChangeDetectionStrategy,
+  HostBinding,
+  ElementRef,
+  AfterViewInit,
+  ChangeDetectorRef,
+} from '@angular/core';
+
+const RADIUS_MAP = {
+  sm: 20,
+  md: 28,
+  lg: 48,
+};
+
+@Component({
+  selector: 'kirby-progress-circle',
+  templateUrl: './progress-circle.component.html',
+  styleUrls: ['./progress-circle.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ProgressCircleComponent implements AfterViewInit {
+  @Input() value: number = 0;
+  @Input() size: 'sm' | 'md' | 'lg';
+  @Input() themeColor: 'success' | 'warning' | 'danger';
+
+  private hasElementBeenVisible = false;
+  private observer: IntersectionObserver;
+
+  constructor(private elementRef: ElementRef, private changeDetectorRef: ChangeDetectorRef) {}
+
+  ngAfterViewInit() {
+    this.observer = new IntersectionObserver(this.onElementVisible);
+    this.observer.observe(this.elementRef.nativeElement);
+  }
+
+  onElementVisible = (entries: IntersectionObserverEntry[]) => {
+    if (entries && entries.length === 1 && entries[0].isIntersecting) {
+      this.observer.unobserve(this.elementRef.nativeElement);
```

Comment:
Consider disconnecting right away:
```suggestion
      this.observer.unobserve(this.elementRef.nativeElement);
      this.observer.disconnect();
```

Reply:
Moved to separate disconnect function which is both called from callback and ngOnDestroy